lib/api_entity.rb

```diff
@@ -45,18 +45,21 @@ def url
     delegate :relationships, to: :class
 
     def inspect
-      if @attributes.blank?
-        candidate_variables = instance_variables - %i[@attributes @casted_by]
-
-        candidate_variables.each_with_object({}) do |variable, acc|
-          key = variable.to_s.sub('@', '')
-          acc[key] = public_send(key)
-        end
-      else
-        keys_to_exclude = relationships.to_a + [:casted_by]
-
-        @attributes.except(*keys_to_exclude)
-      end
+      inspection = if defined?(@attributes) && @attributes.any?
+                     @attributes.except(*(relationships.to_a + [:casted_by]))
+                                .map { |k, v| "#{k}: #{v.inspect}" }
+                                .join(', ')
+                   elsif instance_variables.without(%i[@attributes @casted_by]).any?
+                     instance_variables.without(%i[@attributes @casted_by])
+                                       .map { |k| k.to_s.sub('@', '') }
+                                       .select(&method(:respond_to?))
+                                       .map { |k| "#{k}: #{public_send(k).inspect}" }
+                                       .join(', ')
+                   else
+                     'not initialised'
+                   end
+
+      "#<#{self.class.name.presence || self.class} #{inspection}>"
```

Comment:
Do we also want the object id?

Reply:
Can do - personally that wouldn't mean much to me - I'm more likely to care about the 'id' attribute, eg `goods_nomenclature_item_id` - I was trying to follow the output of ActiveRecord as closely as possible on the theory that theirs must work well or someone would have changed it by now

Reply:
I had a review of the active record inspect implementation. I don't think we'll want it on second thoughts. I've only ever needed it when learning about how ruby works, tbh, and that was a few :moon: :moon: ago